src/Sidekick.Business/Apis/Poe/Trade/Data/Items/ItemDataService.cs

```diff
@@ -61,64 +58,43 @@ private void FillPattern(Rarity? rarity, List<ItemData> items)
-                if (string.IsNullOrEmpty(item.Name))
-                {
-                    if (item.Rarity == Rarity.DivinationCard || item.Rarity == Rarity.Gem)
-                    {
-                        regex = new Regex($"^{Regex.Escape(item.Type)}$");
-                    }
-                    else
-                    {
-                        regex = new Regex(Regex.Escape(item.Type));
-                    }
-                }
-                else
+                var key = item.Name ?? item.Type;
+
+                if (!Patterns.TryGetValue(key, out var itemData))
                 {
-                    regex = new Regex($"^{Regex.Escape(item.Name)}$");
+                    itemData = new List<ItemData>();
+                    Patterns.Add(key, itemData);
                 }
 
-                Patterns.Add((regex, item));
+                itemData.Add(item);
             }
         }
 
-        public ItemData ParseItem(string text)
+        public ItemData ParseItemData(ItemSections itemText)
         {
-            var lines = NewlinePattern.Split(text);
             var results = new List<ItemData>();
 
-            if (lines.Length >= 2)
+            if (Patterns.TryGetValue(itemText.HeaderSection[1], out var itemData))
             {
-                results.AddRange(Patterns
-                    .Where(x => x.Regex.IsMatch(lines[1]))
-                    .Select(x => x.Item)
-                    .ToList());
+                results.AddRange(itemData);
             }
-
-            if (lines.Length >= 3)
+            else if (itemText.HeaderSection.Length > 2 && Patterns.TryGetValue(itemText.HeaderSection[2], out itemData))
             {
-                results.AddRange(Patterns
-                    .Where(x => x.Regex.IsMatch(lines[2]))
-                    .Select(x => x.Item)
-                    .ToList());
+                results.AddRange(itemData);
             }
 
-            // We need to check for vaal gems
-            if (results.Any(x => x.Rarity == Rarity.Gem))
+            if(results.Any(item => item.Rarity == Rarity.Gem)
+                && itemText.TryGetVaalGemName(out var vaalGemName)
+                && Patterns.TryGetValue(vaalGemName, out itemData))
             {
-                foreach (var line in lines.Skip(3))
-                {
-                    results.AddRange(Patterns
-                        .Where(x => x.Regex.IsMatch(line))
-                        .Select(x => x.Item)
-                        .ToList());
-                }
+                // If we find a Vaal gem, we don't care about other matches
+                results.Clear();
+                results.Add(itemData.First());
             }
 
             return results
                 .OrderBy(x => x.Rarity == Rarity.Unique ? 0 : 1)
                 .ThenBy(x => x.Rarity == Rarity.Unknown ? 0 : 1)
-                .ThenByDescending(x => x.Rarity == Rarity.Gem ? x.Text.Length : 0)
                 .FirstOrDefault();
         }
```

Comment:
As we discussed on Discord. This needs to be tested with items that have prefixes and/or suffixes like Blighted and Superior. I don't think it works (untested) by reading the code.

The thing I don't really want is to have to maintain a list of Prefixes or Suffixes like Superior in all languages to be able to strip it. If GGG decide to add a new prefix or suffix, we would need to update our tool.

Reply:
While I agree that it would be nice to avoid having to maintain a list of prefixes and suffixes, I'm not sure the benefit is that big in reality. Aren't all prefixes (or suffixes) that GGG add to the game of interest to us anyway, in terms of parsing and querying? We already keep track of `Blighted` to be able to parse blighted maps, surely we would want to do the same for future league mechanics?

Say they add a new prefix that potentially makes an item more valuable. If we parse the item without handling the prefix at all we could end up makingperform very misleading queries. Price checking a synthesised items will currently not result in correct queries, for example.

Reply:
Of course Blighted and Superior are different in that way. We really don't care about Superior. We do care about Blighted so of course it will be a case by case.

What I want to avoid is something in the lines of
```
input.Replace("Superior ", "")
```

Reply:
I see what you're saying. However, I feel that is a fairly small trade-off in order to be able to use a dictionary for the type instead of 6000 regex matches. If we create a list of the known prefixes we can do a quick loop over them and do the replacements before the dictionary lookup. I don't think it will be that messy to be honest.

@domialex @pobiega Any thoughts on this?

Reply:
I agree with @Blinke. Any new prefixes or suffixes will need to be handled on a case-by-case basis regardless, as Sidekick can't on its own decide if they are relevant or not, so simply adding "Superior" to some "list of prefixes we can safely ignore" feels like a very small cost to improving performance and readability by orders of magnitude.

Reply:
If we go this route, we lose a lot of flexibility. When a new prefix/suffix comes out, we will need to handle it in all languages.

Currently, with Regexes, the key DescriptionEvasion is not translated for every language and nothing breaks dramatically. The only thing would be that the evasion filter is not initialized, which is not that big a deal as you can still show prices and filter.

If we change to a dictionary, I agree that the performance would be better, but very not that much. The Regex for the names and types are super fast compared to the 6000 regexes used for the modifier detection.

I think we could compromise and use Regexes for the types (items that have string.IsNullOrEmpty(item.Name)). Types are the only strings that have stuff added to them such as Blighted or Superior. And use a dictionary for the items that have names, or are divination cards or gems.

I really don't want a solution where if we don't update the client for a new patch/league, we break the application for our users. Also, I will not be able to translate all the prefixes or suffixes as soon as a new release comes out. Waiting for community help to translate can take a while.

Reply:
We basicly have two different approaches to unknown modifiers availble to us; ignore them (the regex way) or abort the operation (the dictionary way).

Lets compare the two approaches pros and cons using a hypothetical scenario. Let's say two new map modifiers are introduced, **"Golden"** and **"Fragile"**. "Golden" increases the value of the map by 10x-ish and "Fragile" doesn't do anything to the market value.

With the regex approach, both fragile and golden maps will parse just fine and return prices based on their base map value, meaning someone might list their 50c map for 5c.

* **+** Works for the irrelevant mod without updating sidekick
* **-** Gives misleading results
* **-** Performance

With the dictionary approach, both fragile and golden maps will result in the result popup having no search results, instead saying "Sidekick couldn't parse that item. Make sure you are using the newest version of sidekick and upload the latest log file to (URL HERE) if you are."

* **+** Performance
* **+** No misleading results
* **-** Requires a Sidekick update whenever new modifiers are created, regardless if the modifier is relevant to pricing or not.

To me, the most important aspect of any "trade helper" is correctness. If you can't trust the values being presented to you, what good are they? Therefor, in my opinion, presenting misleading results is a bigger "break of application" than saying "Sorry, we can't price this item right now".

Reply:
I miss-clicked, didn't mean to resolve yet.
But I agree that it's dangerous to show unreliable results in case of updates, like Blighted for example. I know it's bothersome to maintain this, but it shouldn't happen that often in my opinion.
@Blinke, all good?

Reply:
Yep, sounds good!